environment/src/main/java/com/sequenceiq/environment/environment/validation/cloudstorage/CloudStorageLocationValidator.java

```diff
@@ -0,0 +1,50 @@
+package com.sequenceiq.environment.environment.validation.cloudstorage;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.providerservices.CloudProviderServicesV4Endopint;
+import com.sequenceiq.cloudbreak.client.CloudbreakServiceUserCrnClient;
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
+import com.sequenceiq.cloudbreak.cloud.model.objectstorage.ObjectStorageMetadataRequest;
+import com.sequenceiq.cloudbreak.cloud.model.objectstorage.ObjectStorageMetadataResponse;
+import com.sequenceiq.cloudbreak.util.ValidationResult.ValidationResultBuilder;
+import com.sequenceiq.environment.credential.v1.converter.CredentialToCloudCredentialConverter;
+import com.sequenceiq.environment.environment.domain.Environment;
+
+@Component
+public class CloudStorageLocationValidator {
+
+    @Inject
+    private CloudbreakServiceUserCrnClient cloudbreakClient;
+
+    @Inject
+    private CredentialToCloudCredentialConverter credentialToCloudCredentialConverter;
+
+    public void validate(String userCrn, String storageLocation, Environment environment, ValidationResultBuilder resultBuilder) {
+        String bucketName = getBucketName(storageLocation);
+        CloudCredential cloudCredential = credentialToCloudCredentialConverter.convert(environment.getCredential());
+        ObjectStorageMetadataRequest request = createObjectStorageMetadataRequest(environment.getCloudPlatform(), cloudCredential, bucketName);
+        CloudProviderServicesV4Endopint cloudProviderServicesV4Endopint = cloudbreakClient.withCrn(userCrn).cloudProviderServicesEndpoint();
```

Comment:
we have a bean for the endpoint

Reply:
I saw this but did not bother commenting on it, but I agree with Gergely, using the other client (as a bean) is better